app/assets/javascripts/models/beta/story.js

```diff
@@ -141,11 +146,28 @@ export const toggleStory = (story) => {
   };
 };
 
-const stateFor = (story) => {
-  if (story.state === "unscheduled" && typeof story.estimate === "number") return "unstarted";
-  if (story.state === "unstarted" && isUnestimatedFeature(story)) return "unscheduled";
-  if (!isFeature(story) && story.state === "unscheduled") return "unstarted";
-  return story.state;
+const stateFor = (story, newAttributes, newStory) => {
```

Comment:
I think you could refactor this function 🤔 
Something like:
```js
import { isNull, has } from 'underscore'

const isUnscheduled = ({ state }) => state === status.UNSCHEDULED

const isUnstartedState = (story, newAttributes) => 
  !story._editing.estimate && newAttributes.estimate

const isUnscheduledState = (story, newAttributes) => 
  story._editing.estimate && (
    isNull(newAttributes.estimate) || (
      isUnscheduled(story._editing) && 
      has(newAttributes, 'state') && 
      !isUnscheduled(newAttributes)
    )
  )

const stateFor = (story, newAttributes, newStory) => {
  const { UNSTARTED, UNSCHEDULED } = status;

  if (isEstimable(story)) {
    if (isUnstartedState(story, newAttributes)) return UNSTARTED
    if (isUnscheduledState(story, newAttributes)) return UNSCHEDULED
  }

  return newStory.state;
}
```

Reply:
Oh, sure! This is a brilliant idea! I was thinking in how to refactor all this 'equals' and comparisons, but with this "isNull" and "has" were very nice.